### PR TITLE
ci: add stable promotion workflow and dev deployment

### PR DESCRIPTION
**Pull request by jboix**
## Description

Introduces a new workflow for promoting specific image versions to stable in ECR, followed by an ECS update in production. It also adds an automatic deployment to the development environment after each release

## Changes Made

- Add workflow to tag images as stable in ECR and update ECS service in production.
- Add workflow for automatic deployment to dev after each release.
- Rename `deploy` job to `publish` in `release.yml` for clarity.
- Grant write permissions to issues and PRs in release job.

## Checklist

- [x] I have followed the project's style and contribution guidelines.
- [x] I have performed a self-review of my own changes.
- [x] I have made corresponding changes to the documentation.
- [x] I have added tests that prove my fix is effective or that my feature works.

### CODE REVIEW

**Review by MGaetan89 — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -105,10 +105,16 @@ quality checks and releases.
...
 2. **Release Workflow**
-   When changes are pushed to the `main` branch, this workflow handles versioning and releases using
+   When changes are pushed to `main`, this workflow handles versioning and releases with
    `semantic-release`. It automatically bumps the version, generates release notes, creates a tag,
-   and publishes a Docker image to an Amazon ECR repository.
-
+   and publishes a Docker image to Amazon ECR. This new version is automatically deployed to the
+   development environment.
+
+3. **Production deployment**
```
Comment (MGaetan89) [inline README.md:113]: I find it weird to have a third item here, when line 99 mentions only two workflows. Maybe this should just be a paragraph?

```suggestion
```
And remove the indentation on the paragraph below

File: README.md [hunk 2]
```diff
@@ -105,10 +105,16 @@ quality checks and releases.
...
+   To deploy a specific version to production, manually trigger the `Promote Version to Stable`
```
Comment (MGaetan89) [inline README.md:114]: Maybe add a link to the workflow?
```suggestion
   To deploy a specific version to production, manually trigger the [`Promote Version to Stable`](https://github.com/SRGSSR/pillarbox-event-dispatcher/actions/workflows/deploy.yml)
```

File: README.md [hunk 3]
```diff
@@ -105,10 +105,16 @@ quality checks and releases.
...
+   workflow from the Actions tab, inputting the desired version number (e.g., 1.0.0). This workflow
+   tags the selected version as stable in the ECR, and forces a new deployment on ECS.
+sdf
```
Comment (MGaetan89) [inline README.md:117]: image
```suggestion

```

File: README.md (hunk 2 above)
Comment (jboix) [reply README.md:114]: I will maybe change this workflow later to use the prerelease/latest mechanism so I don't want to delve too much into this documentation now.

File: README.md (hunk 1 above)
Comment (jboix) [reply README.md:112]: I changed the sentence instead.

**Review by MGaetan89 — APPROVED** [review]

**Pull request merged**